docs/mockwbemserver.rst

```diff
@@ -851,50 +853,48 @@ that somewhere an instance the class must be created each time a namespace is
 created or the responder must get the set of namespaces from the CIM repository
 to generate responses to ``GetInstance``, ``EnumerateInstance``, etc.
 
-Also, the :meth:`pywbem_mock.WBEMConnection.InvokeMethod` on
-:class:`pywbem_mock.FakedWBEMConnection` client operation depends on a specific
-responder method in the WBEM server since the behavior of WBEM server methods
-is dependent on the specific method.
+Also, the :meth:`~pywbem_mock.WBEMConnection.InvokeMethod` operation on
+on a :class:`~pywbem_mock.FakedWBEMConnection` object depends on a specific
+responder method in the mock WBEM server.
 
 To meet these needs, pywbem_mock has implemented the capability to create
 :term:`user-defined providers <user-defined provider>` that become request
 responders for defined classes and operations which can can manipulate the
 client input and generate responses specific for defined CIM classes.
 
-User-defined providers can be written by the user by defining subclasses of
-specific provider classes used by the fake WBEM server and registering these
-classes as providers using
+User-defined providers can be written by the user by implementing subclasses of
+specific provider classes and registering these subclasses as providers using
 :meth:`~pywbem_mock.FakedWBEMConnection.register_provider`.
 
-Pywbem defines user-defined provider types so that specific request operations
-can be executed for each provider type.
+The WBEM operations supported by user-defined providers can be implement one by
+one. If a user-defined provider does not implement all WBEM operations, the
+default implementation is used.
 
+The following table shows the WBEM operations for which user-defined providers
+are supported, and the corresponding provider types:
 
-.. table:: Pywbem_mock provider types
+.. table:: WBEM operations of user-defined providers
 
-    ====================== =================== ======================== ================================
-    Provider Type          type name           CIM Request Operations   Default provider class
-    ====================== =================== ======================== ================================
-    method                 "method"            InvokeMethod             :ref:`Method Provider`
-    instance write         "instance-write"    CreateInstance           :ref:`Instance Write Provider`
-    instance write         "instance-write"    ModifyInstance           :ref:`Instance Write Provider`
-    instance write         "instance-write"    DeleteInstance           :ref:`Instance Write Provider`
-    ====================== =================== ======================== ================================
-
-Each user-defined provider is a Python class which is a subclass
-of the corresponding default provider for the provider type as defined in
-the previous table.
+    ============== ================ ============== ==============================
+    Provider type  Type name        WBEM operation Details
```

Comment:
I prefer more than "Details". However, we do have an issue in that we don't have one clear name for the class that is the defining superclass for user-defined providers.   I used "Default provider class". Works for instances but not really for methods.  Something like "API defining class" but that is awkward. How about "Defining class"?

Reply:
The links in that column target the section in the documentation that describes the provider type, so I changed the column heading to "Description".

In these sections, the language also needs to be cleaned up somewhat. Will do that in the next update.